plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/TextualScenario.xtext

```diff
@@ -106,3 +106,13 @@ ElseBlock:
 Block:
 	begin="{" (blockElements+=(Message|Reference|CombinedFragment))* end="}"
 ;
+
+
+StateMode:
+	on='on' timeline=STRING typeKeyword=('state'|'mode') name=STRING
+;
+
+AllocFunction: 
+	on='on' timeline=STRING funcKeyword='function' name=STRING
+;
```

Comment:
Merge the two rules in one rule, they are very similar.
use keyword as variable:

StateFragment:
	on='on' timeline=STRING typeKeyword=('state'|'mode'|'function') name=STRING
;